inst/include/dust2/discrete/system.hpp

```diff
@@ -460,6 +461,13 @@ class dust_discrete {
                         internal,
                         adjoint_next);
       std::swap(adjoint_curr, adjoint_next);
+      for (const auto& el : zero_every) {
```

Comment:
This, and passing `zero_every` through (above) are the substantive changes in this PR